tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -93,12 +92,66 @@ spec:
             echo -n $tags | jq -c
         }
 
-        # Merge the mapping key contents in the data json file with the components key in the snapshot based
-        # on component name. Save the output as a compact json in mapped_snapshot.json file in the workspace
-        { echo -n $(cat "${SNAPSHOT_SPEC_FILE_ORIG}"); echo "${MAPPING}"; } | jq -c -s '.[0] as $snapshot
-          | .[0].components + .[1].components | group_by(.name) | [.[] | select(length > 1)]
-          | map(reduce .[] as $x ({}; . * $x)) as $mergedComponents | $snapshot | .components = $mergedComponents' \
-          > "${SNAPSHOT_SPEC_FILE}"
+        convert_to_quay () { # Convert the repository URL to the quay.io format
+            local repository=$1
+            if [[ $repository == registry.redhat.io/* ]]; then
+                # Convert registry.redhat.io URLs to quay.io/redhat-prod format
+                local new_repo=${repository/registry.redhat.io/quay.io/redhat-prod}
```

Comment:
Why are you using echo and sed for 2 of the 3 substitutions? Why replace `registry.redhat.io/` with `quay.io/redhat-prod/`, change that to `quay.io----redhat-prod----` and then change that to `quay.io/redhat-prod/` instead of just going from `registry.redhat.io/product/repo` to `quay.io----redhat-prod----product----repo` and replace `quay.io----redhat-prod----product` ... you are using 3 substitutions instead of 2. Did you look at how it is done in https://github.com/konflux-ci/release-service-utils/blob/main/utils/translate-delivery-repo? It is much easier to follow there

Reply:
I will consider your suggestion. right now testing locally before pushing the changes here.